extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -750,6 +781,45 @@ export class AmpA4A extends AMP.BaseElement {
   }
 
   /**
+   * Refreshes ad slot by fetching a new creative and rendering it. This leaves
+   * the current creative displayed until the next one is ready.
+   *
+   * @param {function()} refreshEndCallback When called, this function will
+   *   restart the refresh cycle.
+   * @return {Promise} A promise that resolves when all asynchronous portions of
+   *   the refresh function complete. This is particularly handy for testing.
+   */
+  refresh(refreshEndCallback) {
+    return new Promise(resolve => {
+      dev().assert(!this.isRefreshing);
+      this.isRefreshing = true;
+      this.tearDownSlot();
+      this.initiateAdRequest();
+      const promiseId = this.promiseId_;
+      this.adPromise_.then(() => {
+        if (!this.isRefreshing || promiseId != this.promiseId_) {
+          // If this refresh cycle was canceled, such as in a no-content
+          // response case, keep showing the old creative.
+          refreshEndCallback();
+          return;
+        }
+        this.getVsync().mutate(() => {
```

Comment:
I think this should be:
return this.mutateElement(() => {
   this.togglePlaceholder(true);
   return timerFor(this.win)....
});

Note that this delays the timer execution until we are able to mutate.  I believe this is the correct behavior but I thought we had discussed relying on priority to impose the delay?  Essentially the flow would be:

- in refresh initiate ad request
- after ad request, if valid creative (which I'm not sure you're asserting) toggle placeholder and schedule for layout
- ensure that priority is not update to 0 even if an AMP creative (this will ensure 1 second delay is imposed)
- layoutCallback can then execute per usual